contrib/win32/install/build/build-msi.ps1

```diff
@@ -0,0 +1,224 @@
+		/// <param name="nameBytes">The name (within that namespace).</param>
+		/// <param name="version">The version number of the UUID to create; this value must be either
+		/// 3 (for MD5 hashing) or 5 (for SHA-1 hashing).</param>
+		/// <returns>A UUID derived from the namespace and name.</returns>
+		public static Guid Create(Guid namespaceId, byte[] nameBytes, int version)
+		{
+			if (version != 3 && version != 5)
+				throw new ArgumentOutOfRangeException("version", "version must be either 3 or 5.");
+
+			// convert the namespace UUID to network order (step 3)
+			var namespaceBytes = namespaceId.ToByteArray();
+			SwapByteOrder(namespaceBytes);
+
+			// compute the hash of the namespace ID concatenated with the name (step 4)
+			var data = namespaceBytes.Concat(nameBytes).ToArray();
+			byte[] hash;
+			using (var algorithm = version == 3 ? (System.Security.Cryptography.HashAlgorithm)System.Security.Cryptography.MD5.Create() : System.Security.Cryptography.SHA1.Create())
+				hash = algorithm.ComputeHash(data);
+
+			// most bytes from the hash are copied straight to the bytes of the new GUID (steps 5-7, 9, 11-12)
+			var newGuid = new byte[16];
+			Array.Copy(hash, 0, newGuid, 0, 16);
+
+			// set the four most significant bits (bits 12 through 15) of the time_hi_and_version field to the appropriate 4-bit version number from Section 4.1.3 (step 8)
+			newGuid[6] = (byte)((newGuid[6] & 0x0F) | (version << 4));
+
+			// set the two most significant bits (bits 6 and 7) of the clock_seq_hi_and_reserved to zero and one, respectively (step 10)
+			newGuid[8] = (byte)((newGuid[8] & 0x3F) | 0x80);
+
+			// convert the resulting UUID to local byte order (step 13)
+			SwapByteOrder(newGuid);
+			return new Guid(newGuid);
+		}
+
+		private static void SwapByteOrder(byte[] guid)
+		{
+			SwapBytes(guid, 0, 3);
+			SwapBytes(guid, 1, 2);
+			SwapBytes(guid, 4, 5);
+			SwapBytes(guid, 6, 7);
+		}
+
+		private static void SwapBytes(byte[] guid, int left, int right)
+		{
+			var temp = guid[left];
+			guid[left] = guid[right];
+			guid[right] = temp;
+		}
+	}
+}
+"@
+
+
+$SourceDirectory = Resolve-Path -LiteralPath $SourceDirectory
+
+$msBuildDir = &"${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe" -latest -products * -requires Microsoft.Component.MSBuild -property installationPath
+if (0 -ne $LASTEXITCODE -or [string]::IsNullOrWhiteSpace($msBuildDir)) {
+	if ($env:CI) {
+		Write-Host "##vso[task.logissue type=error]MsBuild installation path not found. ErrorCode: $LASTEXITCODE; Error: $msBuildPath"
+	}
+	throw "MsBuild installation path not found. ErrorCode: $LASTEXITCODE; Error: $msBuildPath"
+} else {
+	$msBuildPath = Join-Path -Path $msBuildDir -ChildPath 'MSBuild\Current\Bin\MSBuild.exe'
+	if (-not (Test-Path -Path $msBuildPath)) {
+		$msBuildPath = Join-Path -Path $msBuildDir -ChildPath 'MSBuild\15.0\Bin\MSBuild.exe'
+		if (-not (Test-Path -Path $msBuildPath)) {
+			if ($env:CI) {
```

Comment:
I had a look at the packaging stuff in the PowerShell Github repo. Perhaps we should replace the code here with the stuff from Powershell entirely.

Reply:
As I said I had a look at the PowerShell Modules used in the the PowerShell repo, which support the CI build. And there you'll find numerous places with the same pattern:

Like this

```Powershell
if ($env:TF_BUILD)
{
    Write-Host "##vso[artifact.upload containerfolder=wix;artifactname=wix]$newXmlFileName"
}

throw "Current files to not match  {$FilesWxsPath}"
```

So I'd consider my code not as bad as it might look.